projects/optic/src/commands/capture/capture.ts

```diff
@@ -204,47 +204,54 @@ const getCaptureAction =
       denormalize: false,
     });
 
+    const serverDir =
+      captureConfig.server.dir === undefined
+        ? config.root
+        : captureConfig.server.dir;
+    const timeout = captureConfig.server.ready_timeout || 3 * 60 * 1_000; // 3 minutes
+    const readyInterval = captureConfig.server.ready_interval || 1000;
     // start app
     let app: ChildProcessWithoutNullStreams | undefined;
-    if (!options.serverOverride && captureConfig.server.command) {
-      const serverDir =
-        captureConfig.server.dir === undefined
-          ? config.root
-          : captureConfig.server.dir;
-      const timeout = captureConfig.server.ready_timeout || 3 * 60 * 1_000; // 3 minutes
-      const readyInterval = captureConfig.server.ready_interval || 1000;
-
-      try {
-        app = await startApp(
-          captureConfig.server.command,
-          serverDir,
-          captureConfig.server.ready_endpoint,
-          readyInterval,
-          timeout,
-          proxy.targetUrl
-        );
-      } catch (err) {
-        proxy.stop();
-        logger.error(err);
-        process.exitCode = 1;
-        return;
-      }
-    }
 
-    // make requests
     let errors: any[] = [];
     try {
+      let bailout: Bailout = { didBailout: false, promise: Promise.resolve() };
+      if (!options.serverOverride && captureConfig.server.command) {
+        [app, bailout] = startApp(captureConfig.server.command, serverDir);
+        // If we don't bail out (i.e. the server is still running), we need the promise to be passed down to the next request
+        if (captureConfig.server.ready_endpoint) {
+          await waitForServer(
+            bailout,
+            captureConfig.server.ready_endpoint,
+            readyInterval,
+            timeout,
+            proxy.targetUrl
+          );
+        }
+      }
+      // TODO handle error handling properly
```

Comment:
So this doesn't actually catch errors raised by requests in this block - didn't handle this in this PR but we can handle it separately (not sure off the top of my head for this)